src/main/java/org/folio/dataexp/service/export/strategies/InstancesExportAllStrategy.java

```diff
@@ -0,0 +1,116 @@
+
+  public InstancesExportAllStrategy(ConsortiaService consortiaService, MarcInstanceRecordRepository marcInstanceRecordRepository, HoldingsRecordEntityRepository holdingsRecordEntityRepository, ItemEntityRepository itemEntityRepository, RuleFactory ruleFactory, RuleHandler ruleHandler, RuleProcessor ruleProcessor, ReferenceDataProvider referenceDataProvider, MappingProfileEntityRepository mappingProfileEntityRepository, InstanceWithHridEntityRepository instanceWithHridEntityRepository, ErrorLogService errorLogService, MarcRecordEntityRepository marcRecordEntityRepository, InstanceEntityRepository instanceEntityRepository, EntityManager entityManager, InstanceEntityDeletedRepository instanceEntityDeletedRepository) {
+    super(consortiaService, marcInstanceRecordRepository, holdingsRecordEntityRepository, itemEntityRepository, ruleFactory, ruleHandler, ruleProcessor, referenceDataProvider, mappingProfileEntityRepository, instanceWithHridEntityRepository, errorLogService, marcRecordEntityRepository, instanceEntityRepository, entityManager);
+    this.instanceEntityDeletedRepository = instanceEntityDeletedRepository;
+  }
+
+  @Override
+  public List<MarcRecordEntity> getMarcRecords(Set<UUID> externalIds, MappingProfile mappingProfile, ExportRequest exportRequest) {
+    if (Boolean.TRUE.equals(mappingProfile.getDefault()) || mappingProfile.getRecordTypes().contains(RecordTypes.SRS)) {
+      if (Boolean.TRUE.equals(exportRequest.getDeletedRecords())) {
+        return marcRecordEntityRepository.findByExternalIdInAndRecordTypeIsAndSuppressDiscoveryIs(externalIds, INSTANCE_MARC_TYPE,
+          exportRequest.getSuppressedFromDiscovery());
+      }
+      return marcRecordEntityRepository.findByExternalIdInAndRecordTypeIsAndStateIsAndLeaderRecordStatusNotAndSuppressDiscoveryIs(
+        externalIds, INSTANCE_MARC_TYPE, "ACTUAL", 'd', exportRequest.getSuppressedFromDiscovery());
+    }
+    return new ArrayList<>();
+  }
+
+  @Override
+  protected void processSlices(JobExecutionExportFilesEntity exportFilesEntity, ExportStrategyStatistic exportStatistic, MappingProfile mappingProfile, ExportRequest exportRequest) {
+    var slice = chooseSlice(exportFilesEntity, exportRequest, PageRequest.of(0, exportIdsBatch));
+    updateSliceState(slice);
+    log.info("Slice size for instances export all: {}", slice.getSize());
+    var exportIds = slice.getContent().stream().map(InstanceEntity::getId).collect(Collectors.toSet());
+    log.info("Size of exportIds for instances export all: {}", exportIds.size());
+    createAndSaveMarc(exportIds, exportStatistic, mappingProfile, exportFilesEntity.getJobExecutionId(),
+      exportRequest);
+    while (slice.hasNext()) {
+      slice = chooseSlice(exportFilesEntity, exportRequest, slice.nextPageable());
+      updateSliceState(slice);
+      exportIds = slice.getContent().stream().map(InstanceEntity::getId).collect(Collectors.toSet());
```

Comment:
@obozhko-folio, seems here only ids of instances are needed so it is possible to reduce database load if retrieve only ids . Could you please double check this.